plugin/trino-hive/src/main/java/io/trino/plugin/hive/statistics/MetastoreHiveStatisticsProvider.java

```diff
@@ -149,7 +149,7 @@ public TableStatistics getTableStatistics(
         catch (TrinoException e) {
             if (e.getErrorCode().equals(HIVE_CORRUPTED_COLUMN_STATISTICS.toErrorCode()) && isIgnoreCorruptedStatistics(session)) {
                 log.error(e);
-                return TableStatistics.empty();
+                return createEmptyTableStatisticsWithPartitionColumnStatistics(columns, columnTypes, partitions);
```

Comment:
It looks possible for this method to itself throw exception in some corner case, so I would let this stats corrupted case send empty stats.